public/js/review-searchBar.js

```diff
@@ -0,0 +1,125 @@
+import { CLASS_LIST } from './util/cssClasses.js';
+import { delay } from './util/util.js';
+import { URL, KEYCODE } from './util/data.js';
+
+class SearchBar {
+  constructor({ domElem, inputArea, suggestions, rollingKeywords }) {
+    this.domElem = domElem;
+    this.inputArea = inputArea;
+    this.suggestions = suggestions;
+    this.rollingKeywords = rollingKeywords;
+  }
+
+  registerEvent() {
+    const { HIDDEN } = CLASS_LIST;
+
+    this.inputArea.addEventListener('focus', () => {
+      this.suggestions.classList.remove(HIDDEN);
+      this.rollingKeywords.classList.add(HIDDEN);
+    })
+
+    this.domElem.addEventListener('mouseleave', () => {
+      if(this.inputArea.value === '') {
+        this.inputArea.blur();
+        this.suggestions.classList.add(HIDDEN);
+        this.rollingKeywords.classList.remove(HIDDEN);
+      }
+      if(this.inputArea.value) {
+        this.inputArea.blur();
+        this.suggestions.classList.add(HIDDEN);
```

Comment:
27라인의 if문에 해당하는 코드가 22라인의 else절에 해당하는 걸로 보이는 데 제 생각이 맞다면 아래와 같이 작성해도 동일하지 않을까요?
```suggestion
      this.inputArea.blur();
      this.suggestions.classList.add(HIDDEN);
      if (this.inputArea.value === '') {
        this.rollingKeywords.classList.remove(HIDDEN);
      }
```

Reply:
오!! 맞아요 else의 의도였습니다. 미처 발견하지 못했네욯ㅎㅎㅎㅎㅎ 알려주셔서 감사합니다 😆~~~~~~!!!!!